frontend/packages/shipwright-plugin/src/components/build-list/BuildTable.tsx

```diff
@@ -74,18 +109,53 @@ export const BuildRow: React.FC<RowFunctionArgs<Build>> = ({ obj: build }) => {
       <TableData className={columnClassNames[1]} columnID="namespace">
         <ResourceLink kind="Namespace" name={build.metadata.namespace} />
       </TableData>
-      <TableData className={columnClassNames[2]}>{build.spec?.output?.image}</TableData>
-      <TableData className={columnClassNames[3]}>{build.status?.message}</TableData>
+      {/* <TableData className={columnClassNames[2]}>{build.spec?.output?.image}</TableData>
+      <TableData className={columnClassNames[3]}>{build.status?.message}</TableData> */}
       <TableData className={columnClassNames[4]}>
+        {latestBuildRun ? (
+          <ResourceLink
+            kind={buildRunKindReference}
+            name={latestBuildRun.metadata?.name}
+            namespace={latestBuildRun.metadata?.namespace}
+          />
+        ) : (
+          '-'
+        )}
+      </TableData>
+      <TableData className={columnClassNames[5]}>
+        {latestBuildRun ? <BuildRunStatus buildRun={latestBuildRun} /> : '-'}
+      </TableData>
+      <TableData className={columnClassNames[6]}>
+        {latestBuildRun ? (
+          <Timestamp timestamp={latestBuildRun.metadata?.creationTimestamp} />
+        ) : (
+          '-'
+        )}
+      </TableData>
+      <TableData className={columnClassNames[7]}>
+        {latestBuildRun ? <BuildRunDuration buildRun={latestBuildRun} /> : '-'}
+      </TableData>
+      <TableData className={columnClassNames[8]}>
         <LazyActionMenu context={context} />
       </TableData>
     </>
   );
 };
 
-export const BuildTable: React.FC<TableProps> = (props) => {
+export const BuildTable: React.FC<TableProps> = (props, namespace) => {
   const { t } = useTranslation();
-
+  const buildRunModel = referenceForModel(BuildRunModel);
+  const [buildRuns, buildRunsLoaded, buildRunsLoadError] = useK8sWatchResource<BuildRun[]>({
+    kind: buildRunModel,
+    namespace,
+    isList: true,
+  });
+  if (!buildRuns && buildRunsLoadError) {
+    return <StatusBox label={t('shipwright-plugin~Logs')} loadError={buildRunsLoadError} />;
+  }
+  if (!buildRuns || !buildRunsLoaded) {
+    return <LoadingBox />;
+  }
```

Comment:
That's a nice idea, but imo we should not wait here until all `BuildRuns` are loaded to show the `Builds`.

Instead of that you can pass the loading state to the `Table` and then to the `BuildRow` to show a loading indicator in last `BuildRun` columns. You can also optimize the lookup in the `BuildRow` when you group the `BuildRuns` already here (just once) by their associated `Build`.

For example:

```ts
type CustomData = {
  buildRuns: {
    latestByBuildName: Record<string, BuildRun>;
    loaded: boolean;
    error: Error | undefined;
  }
}

const customData = React.useMemo(() => ({
  buildRuns: {
    latestByBuildName: buildRuns.reduce<Record<string, BuildRun>((acc, buildRun) => {
      const name = buildRun.metadata.labels?.[BUILDRUN_TO_BUILD_REFERENCE_LABEL];
      if (!acc[name] || isBuildRunNewerThen(buildRun, acc[name]) /* compare creationTimestamp in a small utils function */) {
        acc[name] = buildRun;
      }
      return acc;
    }, {}),
    loaded: buildRunsLoaded,
    error: buildRunsLoadError,
  },
}, [buildRuns, buildRunsLoaded, buildRunsLoadError])
```

The main benefit is that you make all this calculation then just once for all BuildRuns.

Instead of once per row component, or in worst case on every row component rerender.

The look up is also easy:

```ts
const latestBuildRun = customData.buildRuns.latestByBuildName[build.metadata.name];
const { loaded, error } = customData.buildRuns;
```

Let me know if you have any question.